pkgs/development/python-modules/pulumi/default.nix

```diff
@@ -1,75 +1,62 @@
 { lib
 , buildPythonPackage
-, fetchpatch
 , fetchFromGitHub
 , protobuf
 , dill
 , grpcio
-, pulumi-bin
+, pulumi
 , isPy27
 , semver
 , pyyaml
 , six
-
-
-# for tests
-, go
-, pulumictl
-, pylint
-, pytest
-, pytest-timeout
-, wheel
-, pytest-asyncio
-
-, mypy
 }:
 let
-  data = import ./data.nix {};
+  # Pulumi uses a pinned version because tests didn't pass with later versions, see
+  # https://github.com/pulumi/pulumi/issues/10301
+  grpcio-1_47 = grpcio.overrideAttrs (finalAttrs: previousAttrs: rec {
+    src = fetchFromGitHub {
+      owner = "grpc";
+      repo = "grpc";
+      rev = "v${version}";
+      hash = "sha256-fMYAos0gQelFMPkpR0DdKr4wPX+nhZSSqeaU4URqgto=";
+      fetchSubmodules = true;
+    };
+
+    version = "1.47.0";
+  });
 in
 buildPythonPackage rec {
+  inherit (pulumi) version src;
+
   pname = "pulumi";
-  version = pulumi-bin.version;
-  disabled = isPy27;
 
-  src = fetchFromGitHub {
-    owner = "pulumi";
-    repo = "pulumi";
-    rev = "v${pulumi-bin.version}";
-    sha256 = "sha256-vqEZEHTpJV65a3leWwYhyi3dzAsN67BXOvk5hnTPeuI=";
-  };
+  disabled = isPy27;
 
   propagatedBuildInputs = [
     semver
     protobuf
     dill
-    grpcio
+    grpcio-1_47
     pyyaml
     six
   ];
 
-  checkInputs = [
-    pulumi-bin
-    pulumictl
-    mypy
-    go
-    pytest
-    pytest-timeout
-    pytest-asyncio
-    wheel
-  ];
+  sourceRoot = "source/sdk/python/lib";
 
-  sourceRoot="source/sdk/python/lib";
   # we apply the modifications done in the pulumi/sdk/python/Makefile
   # but without the venv code
   postPatch = ''
     cp ../../README.md .
-    sed -i "s/\''${VERSION}/${version}/g" setup.py
+    substituteInPlace setup.py --replace "3.0.0" "${version}"
   '';
 
-  # disabled because tests try to fetch go packages from the net
-  doCheck = false;
+  # Verify that the version substitution works
+  checkPhase = ''
+    pip show "${pname}" | grep "Version: ${version}" > /dev/null \
```

Comment:
Replaced `checkPhase`s should have `runHook preCheck` and `runHook postCheck` at the beginning and end of their bodies respectively.

Reply:
Moved to a `preCheck` and enabled tests via `pytest` for the package